netty/src/main/java/io/grpc/netty/NettyClientHandler.java

```diff
@@ -234,11 +252,36 @@ private void onHeadersRead(int streamId, Http2Headers headers, boolean endStream
   /**
    * Handler for an inbound HTTP/2 DATA frame.
    */
-  private void onDataRead(int streamId, ByteBuf data, boolean endOfStream) {
+
+  private void onDataRead(int streamId, ByteBuf data, int padding, boolean endOfStream) {
+    if (!pinging) {
+      pinging = true;
+      sendDataPing(ctx());
+    }
+    dataSizeSincePing += data.readableBytes() + padding;
     NettyClientStream stream = clientStream(requireHttp2Stream(streamId));
     stream.transportDataReceived(data, endOfStream);
   }
 
+  private void sendDataPing(ChannelHandlerContext ctx) {
+    dataSizeSincePing = 0;
+    ChannelPromise promise = ctx.newPromise();
+    long pingData = BDP_MEASUREMENT_PING;
+    ByteBuf buffer = ctx.alloc().buffer(8);
+    buffer.writeLong(pingData);
```

Comment:
There is no need to always create a new ByteBuf. Just make `BDP_MEASUREMENT_PING` a ByteBuf and reuse that. Also I think there might be a problem with endianess when writing a long, instead it might be safer / simpler to write a sequence of bytes / byte array.